### PR TITLE
Fix kraken db loading fromPath

### PR DESCRIPTION
**Pull request by maxibor**
Kraken database loading fixed to `fromPath` method for channel creation

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] `CHANGELOG.md` is updated.

### CODE REVIEW

**Review by apeltzer — APPROVED** [review]

File: main.nf [hunk 1]
```diff
@@ -3108,14 +3108,16 @@ if (params.run_metagenomic_screening && params.database.endsWith(".tar.gz") && p
     path(dbname) into ch_krakendb
     
     script:
-    dbname = params.database.tokenize("/")[-1].tokenize(".")[0]
+    dbname = ckdb.toString() - '.tar.gz'
```
Comment (jfy133) [inline main.nf:3111]: Oooooh clever!

**Review by jfy133 — APPROVED** [review]

**Review by jfy133 — COMMENTED** [review]
CI fail :(

```
Error executing process > 'decomp_kraken'
Caused by:
  Process `decomp_kraken` terminated with an error exit status (1)
Command executed:
  tar xvzf eager_test.tar.gz
  mkdir -p eager_test
  mv *.k2d eager_test
Command exit status:
  1
Command output:
  eager_test/
  eager_test/opts.k2d
  eager_test/seqid2taxid.map
  eager_test/taxo.k2d
  eager_test/hash.k2d
Command error:
  mv: cannot stat '*.k2d': No such file or directory
  ```

**Review by jfy133 — CHANGES_REQUESTED** [review]
See comment above about CI fail

File: main.nf [hunk 2]
```diff
@@ -3108,14 +3108,16 @@ if (params.run_metagenomic_screening && params.database.endsWith(".tar.gz") && p
...
     """
     tar xvzf $ckdb
+    mkdir -p $dbname
+    mv *.k2d $dbname
     """
   }
 
 } else if (! params.database.endsWith(".tar.gz") && params.run_metagenomic_screening && params.metagenomic_tool == 'kraken') {
-    ch_krakendb = path(params.database)
+    ch_krakendb = Channel.fromPath(params.database)
```
Comment (drpatelh) [inline main.nf:3120]: I think you should be ok to use `file()` here instead of `Channel.fromPath()`

**Comment by apeltzer** [comment]
Guessing from the conversation still some fixes required no ?

File: main.nf (hunk 2 above)
Comment (maxibor) [reply main.nf:3120]: I'm passing a directory, so `fromPath` makes more sense

Comment (drpatelh) [reply main.nf:3120]: Nope, `file()` will work equally well. `fromPath` is for globs.

File: main.nf (hunk 1 above)
Comment (maxibor) [reply main.nf:3111]: Kudos to @drpatelh for this one ;)

**Review by jfy133 — APPROVED** [review]
This in principle to me now looks good. CI tests do pass (the one fail is the flakiness on the BAM filtering thing). 

@maxibor please decide if you want to change `fromPath` to `file()` (see Harshil's comment), and then proceed with merging!

**Comment by maxibor** [comment]
As I can't see `file()` anywhere in the doc for channel creation, I think for my future self, it's better to keep `fromPath()` to understand what I did

**Comment by jfy133** [comment]
> As I can't see `file()` anywhere in the doc for channel creation, I think for my future self, it's better to keep `fromPath()` to understand what I did

Feel free to merge!

**Comment by apeltzer** [comment]
Yeah add either `file()` or an additional `collect()` in the respective process to get past this 👍🏻

**Comment by maxibor** [comment]
Went with `.first()` ;) Same effect: transforms a queue channel to value channel (There is anyway only a single item in this channel, the kraken db directory)

File: main.nf (hunk 2 above)
Comment (maxibor) [reply main.nf]: .fromPath().first()

**Comment by apeltzer** [comment]
See slack - solves the issue 👍🏻

**Pull request merged**